src/msw/window.cpp

```diff
@@ -4705,6 +4705,89 @@ wxWindowMSW::MSWOnMeasureItem(int id, WXMEASUREITEMSTRUCT *itemStruct)
     return false;
 }
 
+// ---------------------------------------------------------------------------
+// DPI
+// ---------------------------------------------------------------------------
+
+/*extern*/
+int wxGetSystemMetrics(int nIndex, wxWindow* win)
+{
+    wxTopLevelWindow* tlw = NULL;
+    if ( win )
+    {
+        tlw = wxDynamicCast(wxGetTopLevelParent(win), wxTopLevelWindow);
+    }
+    else
+    {
+        wxWindow* window = static_cast<wxApp*>(wxApp::GetInstance())->GetTopWindow();
+        if ( window )
+        {
+            tlw = wxDynamicCast(wxGetTopLevelParent(window), wxTopLevelWindow);
+        }
+    }
+
+    if ( tlw )
+    {
+        wxDynamicLibrary dllUser32;
+        if ( dllUser32.Load(wxS("User32.dll"), wxDL_VERBATIM | wxDL_QUIET) )
+        {
+            typedef int(WINAPI *GetSystemMetricsForDpi_t)(int nIndex, UINT dpi);
+            GetSystemMetricsForDpi_t wxDL_INIT_FUNC(pfn, GetSystemMetricsForDpi, dllUser32);
```

Comment:
I think this ought to be static, surely we don't want to resolve this function on every call.